### DISCUSSION TITLE
LineString Layer type when reading all layer

### DISCUSSION THREAD

**Discussion by cubecleveland** [geoman-io/leaflet-geoman#1118] [Questions & Help]
Helo,
We have a flow where we add shapes via the editor than loop the shapes to save to DB.
We are using the Uril in geoman to read all the layers.

When reading polygon lines -- In addition to the Polygon feature type we sometime get a LineString feature type in addition to the polygon (so we get 2 layers when there is only 1 on the screen) is that something internal to gEoman when creating poylygon with borders ?

**Comment by Falke-Design** [comment]
Can you please create a demo? https://jsfiddle.net/o1dwu2vg/

**Comment by cubecleveland** [comment]
I just noticed that you are using a different method to retrieve the drawn shaped
   map.pm.Draw.getShapes()

i was using the folliwing loop in my code: 

 mainMapInstance.eachLayer(function (layer) { 
if(layer instanceog L.Marker || layer instanceof L.path)  {

///save shape
})

is your the getShape method screen throught those layer automatically ?

Im working ona  feddle with both options... but wante to know in advance...
Thanks!

**Reply by Falke-Design** [reply]
I don't really understand this sentence:
> is your the getShape method screen throught those layer automatically ?

The shape of a layer should make it possible to difference between custom draw types.

**Comment by cubecleveland** [comment]
FYI, i found the bug on my end -- 

apparently, getShapes screen temporary Line String Layers intenally created by Geoman ? so running it will actually nor return any Line String layers.

However using the following loop to retrive the layers:
    mainMapInstance.eachLayer(function (layer) {
    if(layer instanceog L.Marker || layer instanceof L.path) {
    ///save shape
    })

Actually gets all the temporary items.

Just though ppl would like to know that especailly when trying ot save shapes to a DB .
Thanks!
